packages/@aws-cdk/aws-ecs/lib/instance-protection-handler/index.py

```diff
@@ -0,0 +1,71 @@
+import boto3, os
+
+client = boto3.client('autoscaling')
+
+def on_event(event, context):
+  print(event)
+  request_type = event['RequestType']
+  if request_type == 'Create': return on_create(event)
+  if request_type == 'Update': return on_update(event)
+  if request_type == 'Delete': return on_delete(event)
+  raise Exception("Invalid request type: %s" % request_type)
+
+def on_create(event):
+  asg_name = os.environ['autoscaling_group_name']
+  props = event['ResourceProperties']
+  managed_termination_protection = props['ManagedTerminationProtection'] == 'true'
+  # update NewInstancesProtectedFromScaleIn for ASG depends on managed_termination_protection
+  client.update_auto_scaling_group(
+      AutoScalingGroupName=asg_name,
+      NewInstancesProtectedFromScaleIn=managed_termination_protection 
+  )
+  response = client.describe_auto_scaling_groups(
+      AutoScalingGroupNames=[
+          asg_name,
+      ],
+      MaxRecords=1
+  )
+  asg_arn = response['AutoScalingGroups'][0]['AutoScalingGroupARN']
```

Comment:
> shall we raise exception here in case `len(response) == 0` because of some accidental resource deletion?

Yap. I will add more checks and try/exception later.